python/main.py

```diff
@@ -1,115 +1,117 @@
+    category = cursor.fetchone()
+    # return {"category": category}
+
+    if not category:
+        cursor.execute("INSERT INTO categories (name) VALUES (?)", (name,))
+        conn.commit()
+        category_id = cursor.lastrowid
+    else:
+        category_id = category[0]
+
+    if image:
+        file_location = f"images/{image.filename}"
+        with open(file_location, "wb+") as file_object:
+            file_object.write(image.file.read())
+        image_name = image.filename
+        # cursor.execute("INSERT INTO items (name, category, image_name) VALUES (?, ?, ?)",
+        #            (name, category, image_name))
+        # category id の変更点
+        cursor.execute("INSERT INTO items (name, category_id, image_name) VALUES (?, ?, ?)",
+                   (name, category_id, image_name))
+    else:
+        image_name = "No image"  # デフォルトの画像名
+        cursor.execute("INSERT INTO items (name, category_id) VALUES (?, ?)",
+                       (name, category_id))
+
+    conn.commit()
+    conn.close()
+    # return {"name": name, "category": category, "image_name": image_name}
+    # category id の変更点
+    return {"name": name, "category_id": category_id, "image_name": image_name}
+
+# 保存された商品情報を取得するエンドポイント
+@app.get("/items/", response_model=List[Item])
+def read_items():
+    conn = get_db_connection()
+    items = conn.execute("SELECT * FROM items").fetchall()
+    conn.close()
+    return [dict(item) for item in items]
+
+# 画像を提供するエンドポイント
+@app.get("/images/{filename}", response_class=FileResponse)
+def read_image(filename: str):
+    return FileResponse(path=f"images/{filename}")
 
 # CORSの設定
-origins = [os.environ.get("FRONT_URL", "http://localhost:3000")]
+origins = ["*"]  # ここで適切なオリジンに設定
+
 app.add_middleware(
     CORSMiddleware,
     allow_origins=origins,
-    allow_credentials=False,
-    allow_methods=["GET", "POST", "PUT", "DELETE"],
+    allow_credentials=True,
+    allow_methods=["*"],
     allow_headers=["*"],
 )
 
-# ルートエンドポイント
-@app.get("/")
-def root():
-    return {"message": "Hello, world!"}
+@app.get("/search")
+def search_item(keyword: str):
+    conn = get_db_connection()
+    # itemsから引っ張ってくる場合
+    # items = conn.execute("SELECT name, category_id, image_name FROM items WHERE name LIKE ?", ('%' + keyword + '%',)).fetchall()
+    # conn.close()
+    # # データベースから取得したRowオブジェクトを辞書リストに変換
+    # items_list = [dict(item) for item in items]
+    # return {"items": items_list}
 
-# アイテム追加エンドポイント
-@app.post("/items")
-async def add_item(name: str = Form(...), category: str = Form(...), image: Optional[UploadFile] = None):
-    # アイテム情報のログ出力
-    logger.info(f"Received item: {name}, Category: {category}")
+    # categoriesから引っ張ってくる場合
+    # LIKEを使うと似ている単語が収集される
+    # categories = conn.execute("SELECT id FROM categories WHERE name LIKE ?", ('%' + keyword + '%',)).fetchall()
+    categories = conn.execute("SELECT id, name FROM categories WHERE name = ?", (keyword,)).fetchall()
```

Comment:
本来の想定としては上のlikeを使った文の方が望ましいです。
```suggestion
    categories = conn.execute("SELECT id FROM categories WHERE name LIKE ?", ('%' + keyword + '%',)).fetchall()
```